### ISSUE TITLE
Re-work sensu-agent to use new asset package

### ISSUE THREAD

**Issue by grepory** [sensu/sensu-go#2038]
Requires #2052 
Requires #2072

**Comment by grepory** [comment]
I've added a new `asset` package to sensu-go as a mechanism for managing assets. We need to replace the agent's `assetmanager` package with this. One thing I did not take into account with the asset package are asset sets. We might want to localize that to the `asset` package--allowing uses to install asset sets or individual assets (installing an asset set is just a loop over the assets and calling Get on each of them).

We might want to centralize the runtime environment setup in the asset package as well, but we can look into that when we prepare for implementation.

A few notes from my initial mental modeling of this:

Merging environments specified in Checks and environment variables for assets would be easier if we had a simple mechanism for merging two `[]string` objects where the values are of the form `VAR=value`. This is complicated by the fact that `PATH` may be one of those variables--in which case the value is a :-delimited list of directories, and we want to prepend a directory to the value. This is the case with all asset environment variables.

All assets are a delivery mechanism for a plugin of some kind. All of these (with the exception of extensions--which are managed long-running processes), have a common execution model of:

- set up an execution environment
- execute the plugin and pass into it some form of input
  - all of these accept different inputs, but they are all `[]byte` as far as Sensu is concerned and return []byte as well as a status code.

It would be useful to have a single way to do this across Checks, Handlers, and Mutators. Should we choose to do that work, let's create a subordinate and required ticket prior to working on this.
